kernel/term_typing.mli

```diff
@@ -43,6 +43,4 @@ val infer_declaration : trust:'a trust -> env ->
   'a constant_entry -> Opaqueproof.proofterm Cooking.result
 
 val build_constant_declaration :
-  force:('a -> constr * 'b) ->
-  iter:((constr -> unit) -> 'a -> 'a) ->
-  env -> 'a Cooking.result -> 'a constant_body
+  env -> Opaqueproof.proofterm Cooking.result -> Opaqueproof.proofterm constant_body
```

Comment:
This seems used only internally, why export it?

Reply:
I am not the one who exported it, and I'd be happy to remove it. I suspect there are still people using the trace mechanism of the interpreter to debug, which requires mli declarations.

Reply:
Maybe they should compile without mlis, everything transparent

Reply:
Yep, maybe dune can help there.

Reply:
Anyways, I'm not touching this in this PR.